### PR TITLE
Create pennylane.md

### PR DESCRIPTION
**Pull request by TheKharleeci**
## Proposed changes
 - add documentation for using Pennylane on Lumi
 - here is a link to the [preview page](https://csc-guide-preview.rahtiapp.fi/origin/ft-add-pennylane-app/).

## Checklist before requesting a review

- [x] I have followed the instructions in the [Contributing](https://github.com/CSCfi/csc-user-guide/blob/master/CONTRIBUTING.md) and [Styleguide](https://github.com/CSCfi/csc-user-guide/blob/master/STYLEGUIDE.md) documents.
- [x] My pull request passes all tests.
- [x] I have included a link to the appropriate [preview page](https://csc-guide-preview.rahtiapp.fi/origin/) (select your branch from the list).

### CODE REVIEW

File: docs/apps/pennylane.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+Pennylane is licensed under
+[Apache License 2.0](https://github.com/PennyLaneAI/pennylane/blob/master/LICENSE).
+
+## Usage
+
+To use the default version of Pennylane on LUMI, initialize
+it with:
+
+```bash
+module load pennylane-lightning-kokkos
```
Comment (JMuff22) [inline docs/apps/pennylane.md:37]: What about `module use ...` ?

File: docs/apps/pennylane.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
+---
+tags:
+  - Free
+---
+
+# Pennylane
+
+PennyLane is an open-source software framework for quantum machine learning,
```
Comment (JMuff22) [inline docs/apps/pennylane.md:8]: Perhaps we should emphasize that this is a GPU enabled piece of software?

File: docs/apps/pennylane.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+PennyLane is an open-source software framework for quantum machine learning, 
+quantum chemistry, and quantum computing.
+
+## Available
+
+Currently supported pennylane versions:
+
+| Version | Module                               | LUMI  | Notes           |
+|:--------|:-------------------------------------|:-----:|-----------------|
+| 0.33.0  | `pennylane-lightning-kokkos/0.33.0`  | X     | default version |
```
Comment (JMuff22) [inline docs/apps/pennylane.md:17]: I believe `pennylane-lightning-kokkos>0.32` is deprecated. The kokkos accelerated part of pennylane-lightning is now provided through `pennylane-lightning>=0.33` https://github.com/PennyLaneAI/pennylane-lightning/releases/tag/v0.33.0

Comment (JMuff22) [reply docs/apps/pennylane.md:17]: I think so. Yeah it's confusing. If 0.33 works well then 0.32 and 0.31 could be removed?

File: docs/apps/pennylane.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    #SBATCH --gpus=1
+    #SBATCH --job-name=pennylane-example
+
+    # setting environment variables to specify how the OpenMP threads in a program are bound to processors
+    export OMP_PROC_BIND=spread  
+    export OMP_PLACES=threads
+        
+    module load pennylane-lightning-kokkos
+    python3 simple_test.py
+    ```
```
Comment (joonas-somero) [inline docs/apps/pennylane.md:73]: If you're using the single tab only to render a title for the code block, I'd suggest `title="LUMI"`.

```suggestion
```bash title="LUMI"
#!/bin/bash
#SBATCH --account=<project>
#SBATCH --partition=dev-g
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=1
#SBATCH --mem=128G
#SBATCH --time=0:10:00
#SBATCH --gpus=1
#SBATCH --job-name=pennylane-example

# setting environment variables to specify how the OpenMP threads in a program are bound to processors
export OMP_PROC_BIND=spread  
export OMP_PLACES=threads

module load pennylane-lightning-kokkos
python3 simple_test.py
```
```

File: docs/apps/pennylane.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
+---
+tags:
+  - Free
+---
+
+# Pennylane
+
+PennyLane is an open-source GPU-enabled software framework for quantum machine learning,
```
Comment (JMuff22) [inline docs/apps/pennylane.md:8]: Pennylane is an open-source cross-platform Python library for quantum machine learning, quantum chemistry and quantum computing. Pennylane-lightning is a high-performance state-vector simulator for Pennylane. It is currently GPU enabled through the [Kokkos](https://kokkos.github.io/kokkos-core-wiki/index.html) framework on LUMI.

**Review by JMuff22 — CHANGES_REQUESTED** [review]

File: docs/apps/pennylane.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+module load pennylane-lightning-kokkos
+python3 <file_name>.py
+```
+
+Submit the script with `sbatch <script_name>.sh`
+
+## More information
+
+- [Pennylane documentation](https://docs.pennylane.ai/en/stable/code/qml.html)
```
Comment (JMuff22) [inline docs/apps/pennylane.md:82]: Add `lightning.kokkos` documentation? https://docs.pennylane.ai/projects/lightning/en/stable/lightning_kokkos/device.html or https://docs.pennylane.ai/projects/lightning/en/stable/

**Review by JMuff22 — APPROVED** [review]
LGTM! Thanks for the changes

**Comment by rkronberg** [comment]
Nice! Just a quick comment, the example batch script is claimed to reserve one GPU and _two_ CPU cores. However, I can see only one being requested? Or does this imply virtual cores (i.e. multi-threading)? Maybe clarify still that?

File: docs/apps/pennylane.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+#SBATCH --partition=dev-g
+#SBATCH --nodes=1
+#SBATCH --ntasks-per-node=1
+#SBATCH --mem=128G
+#SBATCH --time=0:10:00
+#SBATCH --gpus=1
+#SBATCH --job-name=pennylane-example
+
+# setting environment variables to specify how the OpenMP threads in a program are bound to processors
+export OMP_PROC_BIND=spread
```
Comment (JMuff22) [inline docs/apps/pennylane.md:76]: If the environment variables for OpenMP are set, this implies that the Kokkos plugin was built with `-DKokkos_ENABLE_OPENMP=ON"` . Perhaps this should be specified. Also whilst many [resources](https://eurotug.github.io/assets/2023_eurotug_slides/KokkosTutorial_EuroTUG_2023_Day1.pdf) online recommend using these environment variables with Kokkos, unless we have explicitly tested their performance on LUMI, I suggest we exclude them.

**Pull request merged**